app/models/server/models/OmnichannelQueue.js

```diff
@@ -0,0 +1,83 @@
+import { Meteor } from 'meteor/meteor';
+
+import { Base } from './_Base';
+
+const UNIQUE_QUEUE_ID = 'queue';
+
+export class OmnichannelQueue extends Base {
```

Comment:
can you convert this into a raw model? this way you create a sync version of `findAndModify` every time `lockQueue` and `unlockQueue` are called.